### ISSUE TITLE
x/tools/gopls: add package selector to completions

### ISSUE THREAD

**Issue by tj** [golang/go#38230]
### What version of Go are you using (`go version`)?

- `go version go1.14 darwin/amd64`
- `golang.org/x/tools/gopls v0.3.4`

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/tj/Library/Caches/go-build"
GOENV="/Users/tj/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOINSECURE=""
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/tj/dev"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/tj/dev/src/github.com/tj/work/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/_d/sf_tjwcj7mj47vvhpkqlhvth0000gn/T/go-build680565461=/tmp/go-build -gno-record-gcc-switches -fno-common"
</pre></details>

### What did you do?

Currently all autocompletion using a package selector will omit the selector. For example if you have a  `t *testing.T`, and type `t.Run` the completed signature will be the following, omitting `t *testing.T`, so you have to manually correct it.

```go
t.Run(name string, f func(t *T))
```

The same happens with any other package, for example while filling fields you'll get `[]Job` in the completion here:

```go
in := client.AddJobsInput{
  Jobs: []Job,
}
```

Where it should be:

```go
in := client.AddJobsInput{
  Jobs: []client.Job,
}
```

(this is in VS Code, if it matters)

**Comment by muirdm** [comment]
This is due to the type alias workaround (https://go-review.googlesource.com/c/tools/+/208497) where we format the *ast.Field.Type expression directly. It has no notion of the types.Qualifier.

**Comment by stamblerre** [comment]
Thanks for the issue report, @tj! We'll aim to get a fix for this into `gopls/v0.5.0`.
